### ISSUE TITLE
Specify notebook_dir when launching the lab

### ISSUE THREAD

**Issue by potoo0** [jupyterlab/jupyterlab#9633]
jlab 2.x support specify notebook_dir when launching the lab, such as `jupyter lab ./`,
but the following command not function in jlab 3.0.5, it is still keep the notebook_dir specified in the global config file.

```bash
jupyter lab ./
jupyter lab --notebook-dir=./
jupyter lab --ServerApp.root_dir=./
```

**Comment by jasongrout** [comment]
It seems you deleted the template that encourages you to go through the trouble-shooting guide (https://jupyterlab.readthedocs.io/en/stable/getting_started/issue.html) and asks for important information to try to understand what might be going on on your computer. Please fill out the template (including error messages, etc.): https://github.com/jupyterlab/jupyterlab/blob/master/.github/ISSUE_TEMPLATE/bug_report.md. It sounds like you may have extra config to let us know about too in order to reproduce the issue.

I just tried, and `jupyter lab --notebook-dir=/some/directory` worked as we would expect.

**Comment by potoo0** [comment]
In my case, arg `--notebook-dir` doesn't work, here is my env:

python:
- os: windows 10 20H2
- python: 3.8.5

jupyter:
```
# pip freeze | grep jupyter
jupyter-client==6.1.7
jupyter-core==4.6.3
jupyter-server==1.2.2
jupyterlab==3.0.5
jupyterlab-code-formatter==1.4.2
jupyterlab-pygments==0.1.1
jupyterlab-server==2.1.2
```

jupyterlab startup command:
```powershell
jupyter lab --notebook-dir=E:\\predict\\3-modelExport
jupyter lab --notebook-dir="E:\\predict\\3-modelExport"
jupyter lab --notebook-dir='E:\\predict\\3-modelExport'
```

startup log:
```
[I 2021-01-20 09:23:26.989 ServerApp] jupyterlab | extension was successfully linked.
[I 2021-01-20 09:23:26.989 ServerApp] jupyterlab_code_formatter | extension was successfully linked.
[W 2021-01-20 09:23:26.993 NotebookApp] 'notebook_dir' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2021-01-20 09:23:26.998 ServerApp] notebook_dir is deprecated, use root_dir
[W 2021-01-20 09:23:27.004 ServerApp] The 'min_open_files_limit' trait of a ServerApp instance expected an int, not the NoneType None.
[I 2021-01-20 09:23:27.020 LabApp] JupyterLab extension loaded from d:\miniconda3\lib\site-packages\jupyterlab
[I 2021-01-20 09:23:27.020 LabApp] JupyterLab application directory is d:\miniconda3\share\jupyter\lab
[I 2021-01-20 09:23:27.023 ServerApp] jupyterlab | extension was successfully loaded.
[I 2021-01-20 09:23:27.024 ServerApp] jupyterlab_code_formatter | extension was successfully loaded.
[I 2021-01-20 09:23:27.202 ServerApp] nbclassic | extension was successfully loaded.
[I 2021-01-20 09:23:27.203 ServerApp] Serving notebooks from local directory: E:\Code\Python\2-advance_tutorial
[I 2021-01-20 09:23:27.203 ServerApp] Jupyter Server 1.2.2 is running at:
[I 2021-01-20 09:23:27.203 ServerApp] http://localhost:8888/lab?token=
```

As shown in the third-to-last line of the log, the notebook_dir is configured in the user config file, instead of the arg specified at startup.

**Comment by potoo0** [comment]
The problem is because the `c.NotebookApp.notebook_dir` in *jupyter_notebook_config.py* overwrites the `notebook-dir` or `ServerApp.root_dir` passed from the command line. If I comment the `c.NotebookApp.notebook_dir` in `jupyter_notebook_config.py`, the dir set by the command line parameter works as expected.
Maybe the notebook_dir/root_dir set via the command line should have the highest priority.

**Comment by jasongrout** [comment]
Is this related? https://github.com/jupyter-server/jupyter_server/pull/400

**Comment by potoo0** [comment]
Solved by upgrade jupyter_server from 1.2.2 to 1.3.0.
